src/main/java/com/cloudbees/jenkins/plugins/bitbucket/hooks/PushHookProcessor.java

```diff
@@ -23,20 +23,30 @@
  */
 package com.cloudbees.jenkins.plugins.bitbucket.hooks;
 
-import java.util.logging.Logger;
-
+import com.cloudbees.jenkins.plugins.bitbucket.api.BitbucketPushEvent;
 import com.cloudbees.jenkins.plugins.bitbucket.client.BitbucketCloudWebhookPayload;
-import com.cloudbees.jenkins.plugins.bitbucket.client.events.BitbucketPushEvent;
+import com.cloudbees.jenkins.plugins.bitbucket.server.client.BitbucketServerWebhookPayload;
+
+import java.util.logging.Logger;
 
 public class PushHookProcessor extends HookProcessor {
 
     private static final Logger LOGGER = Logger.getLogger(PushHookProcessor.class.getName());
 
     @Override
-    public void process(String payload) {
+    public void process(String payload, BitbucketType instanceType) {
         if (payload != null) {
             // TODO: generalize this for BB server
-            BitbucketPushEvent push = BitbucketCloudWebhookPayload.pushEventFromPayload(payload);
+            BitbucketPushEvent push;
+            switch (instanceType)
+            {
```

Comment:
Code style.
And why not a plain `if`? It's not going to be more than 2 options.